Tests/ML/utils/test_io_util.py

```diff
@@ -198,76 +198,108 @@ def write_test_dicom(array: np.ndarray, path: Path) -> None:
     This function DOES NOT create a usable Dicom file and is meant only for testing: tags are set to
     random/default values so that pydicom does not complain when reading the file.
     """
-
     image = sitk.GetImageFromArray(array)
     writer = sitk.ImageFileWriter()
-    writer.KeepOriginalImageUIDOn()
     writer.SetFileName(str(path))
     writer.Execute(image)
 
 
-def test_load_dicom_image_ones(test_output_dirs: TestOutputDirectories) -> None:
+def get_mock_function(monochrome2: bool, bits_stored: Optional[int] = None) -> Callable:
     """
-    Test loading of 2D Dicom images filled with (uint16) ones.
+    SimpleITK does not allow us to set the Photometric Interpretation and Stored Bits tags when writing the Dicom image.
+    For the tests, if an image should set these tags we wrap the reader to return the right attributes.
```

Comment:
I dont understand this sentence

Reply:
SimpleITK always assumes images are of type MONOCHROME2. It can read the MONOCHROME1 images correctly, but when writing for some reason even if we set the tag to be MONOCHROME1 it writes it out as MONOCHROME2. So for the tests, we write an inverted image in MONOCHROME2, and then use a wrapper around the SimpleITK metadata reader to make it look to the test like the tag was MONOCHROME1. Same for the other tag.